plugins/rsyslog.yaml

```diff
@@ -0,0 +1,105 @@
+    default: "UTC"
+    relevant_if:
+      protocol:
+        equals: "rfc3164 (BSD)"
+
+# Set Defaults
+# {{$listen_address := default "0.0.0.0:514" .listen_address}}
+# {{$connection_type := default "udp" .connection_type}}
+# {{$protocol := default "rfc5424 (IETF)" .protocol}}
+# {{$location := default "UTC" .location}}
+
+# Pipeline Template
+pipeline:
+# {{ if eq $connection_type "udp" }}
+  - id: rsyslog_input
+    type: udp_input
+    listen_address: {{ $listen_address }}
+    labels:
+      log_type: rsyslog
+      plugin_id: {{ .id }}
+    output: rsyslog_parser
+# {{ end }}
+
+# {{ if eq $connection_type "tcp" }}
+  - id: rsyslog_input
+    type: tcp_input
+    listen_address: {{ $listen_address }}
+    labels:
+      log_type: rsyslog
+      plugin_id: {{ .id }}
+    output: handle_new_lines
+
+  # tcp input uses \n to break messages into seperate
+  # entries, this can be a problem for Syslog if the
+  # structured_data key=value pairs or message fields contain
+  # a \n. This is unlikly, but not impossible. It has been
+  # observed with rfc5424 structured_data field.
+  # '<' can be relied on to be the first character of every
+  # syslog message for both rfc 5424 and 3164
+  - id: handle_new_lines
+    type: recombine
+    combine_field: $record
+    is_first_entry: "$record startsWith '<'"
+    output: return_router
+# {{ end }}
+
+  - id: return_router
+    type: router
+    default: rsyslog_parser
+    routes:
+      - expr: '$record matches ".*\\\\r.*"'
+        output: return_parser
+
+  - id: return_parser
+    type: regex_parser
+    parse_from: $record
+    regex: '(?P<message1>.*)(?P<return>\\r)(?P<message2>.*)'
+    output: return_restructurer
+
+  - id: return_restructurer
+    type: restructure
+    ops:
+      - add:
+          field: $record
+          value_expr: '$record.message1 + "\\\\r" + $record.message2'
+    output: rsyslog_parser
```

Comment:
The original Syslog implementation needs to handle special situations where return characters were present in the log entry. I do not believe we will run into this issue when targeting rsyslog specifically. This special handling was specifically for NetMotion over syslog. https://github.com/observIQ/stanza-plugins/pull/244

```suggestion

```